### DISCUSSION TITLE
Is there a cron option for certain second?

### DISCUSSION THREAD

**Discussion by ggree1** [Miksus/rocketry#93] [Q&A]
Firstly, thanks for this great project!
One question, can I use cron option to start every minute at certain second like below?

`
@account_sched.scheduled_job("cron", hour="0-23", minute="0-59", second="2", id="test1")
`

I'm using like that with apscheduler, is it possible to use this style with rocektry?

**Comment by Miksus** [comment]
Thanks! It's nice to hear people like it.

Cron (the official specs) has minute as the lowest argument and [Rocketry implemented it to mimic this](https://github.com/Miksus/rocketry/blob/master/rocketry/time/cron.py#L107). The problem is that if *second* is added, then passing the cron expression as string (like ``* * * * *``) either breaks or is inconsistent with the keyword arguments. I'm not sure if the latter is a deal breaker or not. Implementing such could be as little as 15 mins of work.

I think your problem can be handled with this:

```python
from rocketry.conds import time_of_minute, cron

@app.task(cron(hour="0-23", minute="0-59") & time_of_minute.at(2))
def do_things():
    ...
``` 
This runs once a minute (+ the other cron restrictions) and only on 2nd second of a minute. It might take a bit to get grip on how the ``time_of_...`` work but they essentially just check whether the current time is at the given period (in this case between 2nd and 3rd second of a minute using clock). They are useful to combine with conditions that also check whether the task run/finished on the period (ie. ``cron``, ``minutely``, ``daily`` etc.). In other words, use the minimum frequency of your task using ``cron``, ``minutely``, ``daily`` etc. and then further restrict the allowed run time using the ``time_of_...`` conditions. You can also use OR (``|``) to achieve similar results.

I have also made some fixes in the time mechanics of the ``TimeOfMinute`` so after the next release, they should allow more options. Also adding ``secondly`` and ``time_of_second`` in the next release. And also added ``crontime`` so you can use the cron time periods to simply check whether the current time is inside the period (and not that the task has not run as well), useful for complex problems. Hopefully will release the new version in the next few weeks.

The time mechanics are quite complex underneath but you can combine them like lego blocks.

Was this what you were after?

**Reply by ggree1** [reply]
Thanks for the very thoughtful answer that's really helpful.
Maybe in your code `time_of_minute.at(2)`  ..at(2) means like `time_of_minute.between(1,2)` , right? It works then.
Thanks again also I'll be looking forward to the next release.
